### ISSUE TITLE
resolveSync not defined in loader context

### ISSUE THREAD

**Issue by brandonkal** [webpack/webpack#9919]
# Bug report

**What is the current behavior?**

**If the current behavior is a bug, please provide the steps to reproduce.**

**What is the expected behavior?**

Loaders should be able to call `this.resolveSync()` for a synchronous module resolution as documented.

See issue #1634 that was closed without resolution for context.

If this API has been removed and the webpack team does not care to restore it, it should be removed from the documentation.

This is important for loaders that require synchronous resolve. It enables plugins to avoid creating a separate enhanced-resolve instance and attempting to make its behavior match the one used by the webpack compiler.

See:

https://github.com/brandonkal/linaria/blob/27888fa8a3e15cedbfdcf779b18d81ca04af77ee/src/loader.ts#L277-L280

For a runnable example, you can clone that repository and run vscode-jest-tests in the VSCode debugger. You will see that the resolveSync is not defined in the loader's context, so the fallback code is executed.

**Other relevant information:**
webpack version:  4.40.2
Node.js version: v10.15.3
Operating System: MacOS 10.12.6
Additional tools: webpack-dev-server

**Comment by sokra** [comment]
Yep it has been removed in webpack >= 4. You can PR a note to the documentation that it has really been removed. Currently it's only on the list of deprecated methods: https://webpack.js.org/api/loaders/#thisresolvesync

**Comment by SimonAlling** [comment]
> Yep it has been removed in webpack >= 4. You can PR a note to the documentation that it has really been removed. Currently it's only on the list of deprecated methods: https://webpack.js.org/api/loaders/#thisresolvesync

Up until now, I've completely missed the fact that `resolveSync` is listed as deprecated, because I've never read the entire page, only ever clicked or `Ctrl + F`'d my way down to `resolveSync`. I think it would be a great improvement to put some sort of tag right next to each deprecated method, clearly marking it as such.
